inject-groovy/src/main/groovy/io/micronaut/ast/groovy/visitor/GroovyEnumElement.java

```diff
@@ -54,17 +60,42 @@ class GroovyEnumElement extends GroovyClassElement implements EnumElement {
 
     @Override
     public List<String> values() {
-        ClassNode cn = (ClassNode) getNativeType();
-        List<String> values = cn.getFields().stream()
-                .filter((fn -> !fn.getName().equals("MAX_VALUE") && !fn.getName().equals("MIN_VALUE")))
-                .filter(fn -> fn.getType().equals(cn))
-                .map(FieldNode::getName)
-                .collect(Collectors.toList());
-        return Collections.unmodifiableList(values);
+        if (values != null) {
+            return values;
+        }
+        initEnum();
+        return values;
     }
 
     @Override
-    public ClassElement withArrayDimensions(int arrayDimensions) {
-        return new GroovyEnumElement(visitorContext, classNode, getAnnotationMetadata(), arrayDimensions);
+    public List<EnumConstantElement> elements() {
+        if (enumConstants != null) {
+            return enumConstants;
+        }
+        initEnum();
+        return enumConstants;
+    }
+
+    private void initEnum() {
+        values = new ArrayList<>();
+        enumConstants = new ArrayList<>();
```

Comment:
initialise these lists with the size of the number of fields (`getFields()` call) to reduce memory and array resizing

Reply:
@graemerocher `getFields()` return 0 in tests, because EnumConstant is Var, not Field. I see 2 solutions: 

1. Add new loop for calculation count of enum constant elements
2. Recreate final lists like this: 'enumConstants = Collections.unmodifiableList(new ArrayList<>(enumConstants))'

Reply:
ok leave it as is